src/cloud/docker/docker-extend.md

```diff
@@ -82,26 +84,147 @@ In addition to built-in extensions, you can add PHP extensions to the PHP contai
+    
+  ```php?start_inline=1
+    'gd' => [
+        '>=7.0 <=7.3' => [
+            self::EXTENSION_TYPE => self::EXTENSION_TYPE_CORE,
+            self::EXTENSION_OS_DEPENDENCIES => ['libjpeg62-turbo-dev', 'libpng-dev', 'libfreetype6-dev'],
+            self::EXTENSION_CONFIGURE_OPTIONS => [
+                '--with-freetype-dir=/usr/include/',
+                '--with-jpeg-dir=/usr/include/'
+            ],
+        ],
+        '>=7.4' => [
+            self::EXTENSION_TYPE => self::EXTENSION_TYPE_CORE,
+            self::EXTENSION_OS_DEPENDENCIES => ['libjpeg62-turbo-dev', 'libpng-dev', 'libfreetype6-dev'],
+            self::EXTENSION_CONFIGURE_OPTIONS => [
+                '--with-freetype=/usr/include/',
+                '--with-jpeg=/usr/include/'
+            ],
+        ],
+
+    ],
+    'geoip' => [
+        '>=7.0' => [
+            self::EXTENSION_TYPE => self::EXTENSION_TYPE_PECL,
+            self::EXTENSION_OS_DEPENDENCIES => ['libgeoip-dev', 'wget'],
+            self::EXTENSION_PACKAGE_NAME => 'geoip-1.1.1',
+        ],
+    ],
+  ```
+   
+  If the installation requires the execution of special commands, then use the type `EXTENSION_TYPE_INSTALLATION_SCRIPT`.
+  The template for this type:
+   
+  ```php?start_inline=1
+      'extension name' => [
+           'php version constraint' => [
+               self::EXTENSION_TYPE => self::EXTENSION_TYPE_INSTALLATION_SCRIPT,
+               self::EXTENSION_INSTALLATION_SCRIPT => <<< BASH
+       <installation script>
+       BASH
+       ],
+  ```
+   
+  Usage example:
+  ```php?start_inline=1
+    'ioncube' => [
+        '>=7.0 <=7.3' => [
+            self::EXTENSION_TYPE => self::EXTENSION_TYPE_INSTALLATION_SCRIPT,
+            self::EXTENSION_INSTALLATION_SCRIPT => <<< BASH
+    cd /tmp
+    curl -O https://downloads.ioncube.com/loader_downloads/ioncube_loaders_lin_x86-64.tar.gz
+    tar zxvf ioncube_loaders_lin_x86-64.tar.gz
+    export PHP_VERSION=$(php -r "echo PHP_MAJOR_VERSION.'.'.PHP_MINOR_VERSION;")
+    export PHP_EXT_DIR=$(php-config --extension-dir)
+    cp "./ioncube/ioncube_loader_lin_\${PHP_VERSION}.so" "\${PHP_EXT_DIR}/ioncube.so"
+    rm -rf ./ioncube
+    rm ioncube_loaders_lin_x86-64.tar.gz
+    BASH
+    ],
+  ```
+1. If you want the extension to be enabled by default, then add it to the array `DEFAULT_PHP_EXTENSIONS`
+
+```php?start_inline=1
+    /**
+     * Extensions which should be installed by default
+     */
+    public const DEFAULT_PHP_EXTENSIONS = [
+        'bcmath',
+        'bz2',
+        'calendar',
+        'exif',
+        'gd',
+        'gettext',
+        'intl',
+        'mysqli',
+        'pcntl',
+        'pdo_mysql',
+        'soap',
+        'sockets',
+        'sysvmsg',
+        'sysvsem',
+        'sysvshm',
+        'opcache',
+        'zip',
+    ];
+```
+Otherwise, to enable, you will need to specify the name of the extension in the file `magento.app.yaml` of your projects and restart docker container.
```

Comment:
```suggestion
Otherwise, to enable an extension, you will need to specify the name of the extension in the file `magento.app.yaml` of your projects and restart docker container.
```